zerver/lib/zcommand.py

```diff
@@ -32,4 +45,23 @@ def process_zcommands(content: str, user_profile: UserProfile) -> Dict[str, Any]
         ret = dict(msg=msg)
         return ret
 
+    if command == 'mute_topic':
+        linked_stream = possible_linked_stream_names(content)
+        if len(tokens) < 3 or len(linked_stream) != 1:
+            return dict(msg="Usage: /mute_topic #<stream_name> <topic_name>")
+
+        [stream_name] = linked_stream
+        topic = get_topic_from_mute_topic_content(content, stream_name)
+        try:
+            stream = get_realm_stream(stream_name=stream_name,
+                                      realm_id=user_profile.realm.id)
+        except Stream.DoesNotExist:
+            return dict(msg='A valid stream name is required.')
+
+        if topic_exists(topic, get_stream_recipient(stream.id)):
+            ret = dict(subject=topic, stream=stream_name, type='stream')
```

Comment:
Is there a reason you are passing `type`? This seems unnecessary.

Reply:
The data passed in [toggle_mute](https://github.com/zulip/zulip/blob/master/static/js/muting_ui.js#L179-L185) requires a `type='stream'`.